### PR TITLE
fix(peer-deps): support `@angular/compiler@10.1`

### PR DESCRIPTION
**Pull request by fisker**

### CODE REVIEW

File: tests/trasnform.test.ts [hunk 1]
```diff
@@ -102,7 +102,13 @@ describe.each`
     expect(beforeNode).not.toEqual(null);
     expect(afterNode).not.toEqual(null);
 
-    expect(getNgType(beforeNode!)).toEqual(beforeType);
+    // `Unary` was `Binary` before `v10.1.0"`
```
Comment (ikatyang) [inline tests/trasnform.test.ts:105]:
```suggestion
    // `Unary` was `Binary` in `@angular/compier@<10.1.0"`
```

File: src/transform.ts [hunk 2]
```diff
@@ -47,10 +47,28 @@ export const transform = (
...
+      return _c<b.UnaryExpression>(
+        'UnaryExpression',
+        {
+          prefix: true,
+          argument: tArgument,
+          operator: isPrefixMinus ? '-' : '+',
+        },
+        node.span,
+        { hasParentParens: isInParentParens },
+      );
```
Comment (ikatyang) [inline src/transform.ts:64]: `left` and `right` seem to be only used for backward compatibility, and it seem to be not recommended to use as they typed them as `never` in their type, which we should probably avoid using them here:

```suggestion
      // @ts-ignore: there is no `Unary` in `@angular/compiler@<10.1.0` 
      const { operator, expr } = node as ng.Unary;
      const tArgument = _t<b.Expression>(expr);
      return _c<b.UnaryExpression>(
        'UnaryExpression',
        {
          prefix: true,
          argument: tArgument,
          operator: operator as '-' | '+',
        },
        node.span,
        { hasParentParens: isInParentParens },
      );
```

File: src/utils.ts [hunk 3]
```diff
@@ -125,6 +129,8 @@ function extractComments(
 
 // prettier-ignore
 export function getNgType(node: (ng.AST | RawNGComment) & { type?: string }) {
+  // @ts-ignore
```
Comment (ikatyang) [inline src/utils.ts:132]:
```suggestion
  // @ts-ignore: there is no `Unary` in `@angular/compiler@<10.1.0`
```

File: jest.config.js [hunk 4]
```diff
@@ -11,7 +12,7 @@ module.exports = {
   testMatch: ['**/*.test.ts'],
   transform: { '\\.ts$': 'ts-jest' },
   coverageReporters: ['lcov', 'text-summary'],
-  collectCoverage: !!process.env.CI,
+  collectCoverage,
```
Comment (ikatyang) [inline jest.config.js:15]: Manually specify which version to collect coverage on CI would be better I think:

```suggestion
  collectCoverage: !!process.env.ENABLE_COVERAGE,
```

```diff
 # .travis.yml
-yarn run test
+ENABLE_COVERAGE=1 yarn run test
```

**Comment by ikatyang** [comment]
Since `@angular/compiler@10.1.0` is included in the current supported range (`^10.0.0`), I'm going to mark this PR as a bug fix instead of a feature.

**Comment by fisker** [comment]
All done.

**Review by ikatyang — APPROVED** [review]

**Comment by ikatyang** [comment]
Thanks!

**Comment by ikatyang** [comment]
👉 https://github.com/ikatyang/angular-estree-parser/releases/tag/v2.2.1

**Pull request merged**